tensorflow/core/kernels/stack.cc

```diff
@@ -61,7 +66,9 @@ class Stack : public ResourceBase {
       return errors::InvalidArgument("Stack[", stack_name_, "] overflowed ",
                                      "its max_size (", max_size_, ")");
     }
-    stack_.push_back(value);
+    int index = stack_.size() - 1;
+    unswapped_lru_.push_back(index);
+    stack_.push_back({value, unswapped_lru_.end()});
```

Comment:
The issue here is unswapped_lru_.end(), which doesn't point to a valid element. It's basically the iterator pointing to element unswapped_lru_[unswapped_lru_.size()], and is only used to check whether you've hit the end of the std::list.

You can use unswapped_lru_.rbegin() here instead.

Reply:
I've pushed the latest commit, and quick tests show it fixes the core dumped error.

Turns out std::list.erase doesn't accept reverse_iterator, so it needs to be converted to forward iterator with `std::next(unswapped_lru_.rbegin()).base()` (https://stackoverflow.com/a/1830240).

This quick test shows it works:

```
#include <bits/stdc++.h>

int main() {
    std::list<int> l;
    l.push_back(1);
    l.push_back(2);
    auto iter = std::next(l.rbegin()).base();
    l.push_back(3);
    l.erase(iter);
    for (auto i = l.begin(); i != l.end(); i++) {
        std::cout << *i << std::endl;
    }
}
```

Also, it seems that checking if the iterator's still valid with `iter != unswapped_lru_.end()` isn't sufficient, so I changed that to this:

```
...
for (auto i = unswapped_lru_.begin(); i != unswapped_lru_.end(); i++) {
  if (i == iter) {
    unswapped_lru_.erase(iter);
    break;
  }
}
...
```